test/gtest/mhp/sort.cpp

```diff
@@ -0,0 +1,54 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+#include <dr/detail/communicator.hpp>
+#include <dr/mhp/algorithms/sort.hpp>
+
+using T = int;
+using DV = dr::mhp::distributed_vector<T, dr::mhp::default_allocator<T>>;
+using V = std::vector<T>;
+
+TEST(MhpSort, Sort) {
+  std::vector<std::size_t> sizes = {
+      1, comm_size - 1, (comm_size - 1) * (comm_size - 1), 4, 7, 10, 23};
+
+  for (std::size_t n : sizes) {
+    std::cout << "Sort: Test size " << n << std::endl;
+    V l_v = generate_random<T>(n, 100);
+    DV d_v(n);
+    for (std::size_t idx = 0; idx < n; idx++) {
+      d_v[idx] = l_v[idx];
+    }
+    barrier();
+
+    std::sort(l_v.begin(), l_v.end());
+    dr::mhp::sort(d_v);
+
+    barrier();
+    EXPECT_TRUE(equal(l_v, d_v));
+  }
+}
+
+TEST(MhpSort, Sort_reverse) {
```

Comment:
This test is a copy of above one.
loops/logic are not warmly welcomed in perfect unit-tests too
Copy-pastes in tests are more allowed as tests has to be readable, but here it is just one param.
I would create helper func `test_sort_randomly(size, functor)` and make your tests look sth like this
```
TEST(MhpSort, random_sort)
{
test_sort_randomly(1, std::less<T>());
test_sort_randomly(comm_size-1, std::less<T>());
test_sort_randomly(4, std::less<T>());
test_sort_randomly(10, std::less<T>());
}

TEST(MhpSort, random_reverse_sort)
{
test_sort_randomly(1, std::greater<T>());
test_sort_randomly((comm_size-1)*(comm_size-1), std::greater<T>());
test_sort_randomly(7, std::greater<T>());
test_sort_randomly(23, std::greater<T>());
}
```

Reply:
Redundant elements moved to helper fuctions. Tests for std::less and std::greater can be automatic, no need to list them separately.